### DISCUSSION TITLE
How to use your functionality to convert the json result I have to csv?

### DISCUSSION THREAD

**Discussion by PrathikshaBallal** [some-natalie/ghas-to-csv#17] [Q&A] [answered]
I wrote python code to output code scanning alerts from all repositories under our Org. Now how do I use this functionality to convert JSON to CSV? Where can I pass my response json?

` def write_repo_cs_list(cs_list):
    """
    Write a list of code scanning alerts to a CSV file.
    Inputs:
    - List of code scanning alerts
    Outputs:
    - CSV file of code scanning alerts
    """

    # Write code scanning alerts to csv file
    with open("cs_list.csv", "w") as f:
        writer = csv.writer(f)
        writer.writerow(
            [
                "number",
                "created_at",
                "html_url",
                "state",
                "fixed_at",
                "dismissed_by",
                "dismissed_at",
                "dismissed_reason",
                "rule_id",
                "rule_severity",
                "rule_tags",
                "rule_description",
                "rule_name",
                "tool_name",
                "tool_version",
                "most_recent_instance_ref",
                "most_recent_instance_state",
                "most_recent_instance_sha",
                "instances_url",
            ]
        )
        for cs in cs_list:
            if cs["state"] == "open":
                cs["fixed_at"] = "none"
                cs["dismissed_by"] = "none"
                cs["dismissed_at"] = "none"
                cs["dismissed_reason"] = "none"
            writer.writerow(
                [
                    cs["number"],
                    cs["created_at"],
                    cs["html_url"],
                    cs["state"],
                    cs["fixed_at"],
                    cs["dismissed_by"],
                    cs["dismissed_at"],
                    cs["dismissed_reason"],
                    cs["rule"]["id"],
                    cs["rule"]["severity"],
                    cs["rule"]["tags"],
                    cs["rule"]["description"],
                    cs["rule"]["name"],
                    cs["tool"]["name"],
                    cs["tool"]["version"],
                    cs["most_recent_instance"]["ref"],
                    cs["most_recent_instance"]["state"],
                    cs["most_recent_instance"]["commit_sha"],
                    cs["instances_url"],
                ]
            )`

**Comment by some-natalie** [comment, marked as answer]
Hey @PrathikshaBallal , you can just use the Action as directed in the [README.md](https://github.com/some-natalie/ghas-to-csv) file.  It uploads a CSV as a workflow artifact.  If you're wanting to do your own parsing of the API, the docs are [here](https://docs.github.com/en/enterprise-cloud@latest/rest) and the docs for Python's CSV library are [here](https://docs.python.org/3/library/csv.html).  Cheers!
